x-pack/plugins/apm/server/lib/services/get_service_metadata_icons.ts

```diff
@@ -29,31 +31,46 @@ interface ServiceMetadataIcons {
   cloudProvider?: string;
 }
 
+export const should = [
+  { exists: { field: CONTAINER_ID } },
+  { exists: { field: POD_NAME } },
+  { exists: { field: CLOUD_PROVIDER } },
+  { exists: { field: HOST_OS_PLATFORM } },
+  { exists: { field: AGENT_NAME } },
+];
+
 export async function getServiceMetadataIcons({
   serviceName,
   setup,
+  searchAggregatedTransactions,
 }: {
   serviceName: string;
   setup: Setup & SetupTimeRange;
+  searchAggregatedTransactions: boolean;
 }): Promise<ServiceMetadataIcons> {
   const { start, end, apmEventClient } = setup;
 
   const filter = [
     { term: { [SERVICE_NAME]: serviceName } },
-    { term: { [PROCESSOR_EVENT]: ProcessorEvent.transaction } },
     { range: rangeFilter(start, end) },
-    ...setup.esFilter,
   ];
 
   const params = {
     apm: {
-      events: [ProcessorEvent.transaction],
+      events: [
+        getProcessorEventForAggregatedTransactions(
+          searchAggregatedTransactions
+        ),
+        ProcessorEvent.error,
+        ProcessorEvent.span,
+        ProcessorEvent.metric,
+      ],
     },
     terminateAfter: 1,
     body: {
       size: 1,
       _source: [KUBERNETES, CLOUD_PROVIDER, CONTAINER_ID, AGENT_NAME],
-      query: { bool: { filter } },
+      query: { bool: { filter, should } },
```

Comment:
I'm not sure how useful a `should` clause is when using `terminateAfter`.

Reply:
@dgieselaar  Yeah, that makes sense, maybe I should remove `terminateAfter` and add `size:1`, then I'd guarantee that the first returned document might have the fields. And it'd be in sync with `metadata/details` API.  WDYT?

Reply:
Yeah sounds good, I don't really see a good alternative here.